core/src/km_core_state_api.cpp

```diff
@@ -262,7 +262,7 @@ void km_core_state_imx_deregister_callback(km_core_state *state)
 }
 
 bool is_context_valid(km_core_cp const * context, km_core_cp const * cached_context) {
-  if (context == nullptr || cached_context == nullptr || *cached_context == NULL) {
+  if (context == nullptr || cached_context == nullptr || *cached_context == '\0') {
```

Comment:
```suggestion
  if (context == nullptr || cached_context == nullptr || *cached_context == 0) {
```

I think that should work in all compilers